include/rgf/README.md

```diff
@@ -0,0 +1,168 @@
+
+7. [References](#7-references)
+
+# 1. Introduction
+
+This software package provides implementation of regularized greedy forest
+(RGF) described in [[1]](#7-references).
+
+## 1.1. System Requirements
+
+Executables are provided only for some versions of Windows
+(see [Section 3](#3-creating-the-executable)) for detail).
+If provided executables do not work for your environment,
+you need to compile C++ code.
+
+To use the provided tools and to go through the examples in the user guide,
+Perl is required.
+
+# 2. Download and Installation
+
+Download the package and extract the content.
+
+The top directory of the extracted content is `rgf`. Below all the
+path expressions are relative to `rgf`.
+
+# 3. Creating the Executable
+
+To go through the examples in the user guide, your executable needs to be
+at the `bin` directory. Otherwise, your executable can be anywhere you like.
+
+## 3.1. Windows
+
+Executable files are provided with the filename `rgf.exe` for 64-bit Windows
+and `rgf32.exe` for 32-bit Windows at the `bin` directory.
+They were tested on Windows 10 with the latest updates as
+of 8/12/2017. You can either use them
+(rename `rgf32.exe` to `rgf.exe` in case you are using 32-bit Windows),
+or you can rebuild them by yourself using the provided solution file for
+MS Visual C++ 2010 Express: `Windows\rgf\rgf.sln`.
+
+## 3.2. Unix-like Systems
+
+You need to build your executable from the source code. A make file
+`makefile` is provided at the `build` directory. It is configured to use
+**g++** and always compile everything. You may need to customize `makefile`
+for your environment.
+
+To build the executable, change the current directory to the `build`
+directory and enter in the command line `make`. Check the
+`bin` directory to make sure that your new executable `rgf` is there.
+
+## 3.3. [Optional] Endianness Consideration
+
+The models obtained by RGF training can be saved to files.
+The model files are essentially snap shots of memory that include
+numerical values. Therefore, the model files are sensitive to
+"endianness" of the environments. For this reason, if you wish to
+share model files among environments of different endianness, you need
+to follow the instructions below. Otherwise, you can skip this section.
+
+To share model files among the environments of different endianness,
+build your executable for the environment with big-endian with the
+compile option:
+
+```
+/D_AZ_BIG_ENDIAN_
+```
+
+By doing so, the executable in your big-endian environment swaps the
+byte order of numerical values before writing and after reading the
+model files.
+
+# 4. Documentation
+
+[`rgf-guide.pdf`](./rgf-guide.pdf) "**Regularized Greedy Forest Version 1.2: User Guide**" is included.
+
+# 5. Contact
+
+Please post an [issue](https://github.com/RGF-team/rgf_python/issues)
+at GitHub repository for any errors you encounter.
+
+# 6. Copyright
+
+RGF Version 1.3 is distributed under the **MIT license**. Please read
```

Comment:
Considering that the name of this repository will become `rgf`, it is confusing.
How about remove version description? And implicitly synchronizing versions with `rgf`.

Reply:
I totally against this idea! Each project should has own versioning.

I don't see any problems with rgf repo name. A lot of implementations could be in it...